### PR TITLE
Block upload if size exceeds capacity

### LINKED ISSUE #2030: Show an informative message when selecting files that will reach the storage limit
As we know the upload size before the "upload" button is clicked, we should warn users about it, and redirect them to the form to get to the early adopter users, able to subscribe to paid plans.

### PR DESCRIPTION
**Pull request by FSM1**
closes #2030

---
Submission checklist: 

#### Layout
- [x] Change looks good in the desktop web ui
- [x] Change looks good in the mobile web ui

#### Theme
- [x] Components / elements inspected in light mode
- [ ] Components / elements inspected in dark mode

#### Billing
- [x] Change applied to both files-ui and storage-ui

### CODE REVIEW

File: packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx [hunk 1]
```diff
@@ -122,49 +134,51 @@ const UploadFileModule = ({ modalOpen, close }: IUploadFileModuleProps) => {
...
-              type="submit"
-              variant="primary"
-              className={clsx(classes.okButton, "wide")}
-              disabled={isDoneDisabled}
-            >
-              <Trans>Start Upload</Trans>
-            </Button>
-          </footer>
-        </Form>
+        {({ isValid }) => (
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx]: This construct with the valid is really weird. Can we use the notation with the hook

```js
const formik = useFormik({
    initialValues:{}
    // ...
  })
  
<FormikProvider value={formik}>
   <Button
     //...
      disabled={isDoneDisabled || !formik.isValid}
    >
</FormikProvider>
```

**Comment by asnaith** [comment]
I'm not able to upload enough files to test this at the moment but I check again later once the current upload issue is resolved.

Hopefully, I can intercept and stub the capacity in charles and avoid the need to upload 20gb worth of data.

File: packages/files-ui/src/Components/Modules/FileBrowsers/SharedFileBrowser.tsx [hunk 2]
```diff
@@ -201,20 +201,31 @@ const SharedFileBrowser = () => {
 
   const handleUploadOnDrop = useCallback(async (files: File[], fileItems: DataTransferItemList, path: string) => {
     if (!bucket) return
-    if (accountRestricted) {
+    if (accountRestricted && bucket.permission === "owner") {
```
[the remark this answers is not in the document]
Comment (FSM1) [reply packages/files-ui/src/Components/Modules/FileBrowsers/SharedFileBrowser.tsx:204]: The upload will still fail, but will only do so once the API throws not before the upload starts. We have no way of knowing how much capacity the owner has on their account, so the UX here could use some improvement.

File: packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx (hunk 1 above)
Comment (FSM1) [reply packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx:137]: > I'm not able to upload enough files to test this at the moment but I check again later once the current upload issue is resolved.
> 
> Hopefully, I can intercept and stub the capacity in charles and avoid the need to upload 20gb worth of data.

should be sufficient enough to add an interceptor on the GET `bucket summary` request in the tests

**Comment by FSM1** [comment]
> I'm not able to upload enough files to test this at the moment but I check again later once the current upload issue is resolved.
> 
> Hopefully, I can intercept and stub the capacity in charles and avoid the need to upload 20gb worth of data.

Should be easy enough to intercept the GET `bucket summary` call in the tests and test it out that way.

**Review by tanmoyAtb — COMMENTED** [review]
Looks great ! Tried all the ways we can upload. 

Made me think about this.
Whats our policy if the shared folder owner is not the current user ?
I see that shared users with edit access can upload without any restriction.

**Comment by tanmoyAtb** [comment]
> Looks great ! Tried all the ways we can upload.
> 
> Made me think about this. Whats our policy if the shared folder owner is not the current user ? I see that shared users with edit access can upload without any restriction.

Ahh I saw the reply later. Makes perfect sense.

**Review by tanmoyAtb — APPROVED** [review]

File: packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx (hunk 1 above)
Comment (FSM1) [reply packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx]: Fixed

File: packages/files-ui/src/Components/Modules/FileBrowsers/CSFFileBrowser.tsx [hunk 3]
```diff
@@ -169,12 +169,23 @@ const CSFFileBrowser: React.FC<IFileBrowserModuleProps> = () => {
...
     }
 
+    const availableStorage = storageSummary?.available_storage || 0
+    const uploadSize = files?.reduce((total: number, file: File) => total += file.size, 0) || 0
+
+    if (uploadSize > availableStorage) {
+      addToast({
+        type: "error",
+        title: t`Upload size exceeds plan capacity`,
+        subtitle: t`Please select fewer files to upload`
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/CSFFileBrowser.tsx:179]: adjust copy

File: packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx [hunk 4]
```diff
@@ -83,8 +83,19 @@ const UploadFileModule = ({ modalOpen, close }: IUploadFileModuleProps) => {
...
-  const UploadSchema = object().shape({ files: array().required(t`Please select a file to upload`) })
+  const UploadSchema = useMemo(() => object().shape({
+    files: array().required(t`Please select a file to upload`)
+      .test("Upload Size",
+        t`Upload size exceeds plan capacity`,
+        (files) => {
+          const isOwner = bucket?.permission === "owner"
+          const availableStorage = storageSummary?.available_storage || 0
+          const uploadSize = files?.reduce((total: number, file: File) => total += file.size, 0) || 0
+          return !(isOwner && uploadSize > availableStorage)
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx]: I find it easier to read with an early return
```suggestion
          // no validation if the user isn't the owner
          if(bucket?.permission !== "owner") return true
          
          const availableStorage = storageSummary?.available_storage || 0
          const uploadSize = files?.reduce((total: number, file: File) => total += file.size, 0) || 0
          return uploadSize < availableStorage
```

File: packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx [hunk 5]
```diff
@@ -117,10 +134,8 @@ const UploadFileModule = ({ modalOpen, close }: IUploadFileModuleProps) => {
         inner: classes.modalInner
       }}
     >
-      <Formik
-        initialValues={{ files: [] }}
-        validationSchema={UploadSchema}
-        onSubmit={onSubmit}
+      <FormikProvider
+        value={formik}
       >
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx]:
```suggestion
      <FormikProvider
        value={formik}
      >
```
```suggestion
      <FormikProvider value={formik}>
```

Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx:139]:
```suggestion
      <FormikProvider value={formik}>
```

File: packages/common-components/src/FileInput/FileInput.tsx [hunk 6]
```diff
@@ -51,6 +51,9 @@ const useStyles = makeStyles(({ constants, palette, overrides }: ITheme) =>
     },
     error: {
       color: palette.error.main,
+      backgroundColor: palette.additional["gray"][3],
+      paddingLeft: 40,
+      paddingBottom: 4,
```
Comment (Tbaut) [inline packages/common-components/src/FileInput/FileInput.tsx]: To have the error vertically centered (my suggestion adds a 4px top and a 40px right that shouldn't matter)
```suggestion
      padding: "4px 40px",
```

**Review by Tbaut — APPROVED** [review]
Nice, it's working well. Just left a couple suggestions for readability and a small layout thing.

**Pull request merged**